src/test/java/se/liu/ida/hefquin/engine/data/utils/SolutionMappingsIterableOverCollectionOfListsTest.java

```diff
@@ -0,0 +1,110 @@
+
+public class SolutionMappingsIterableOverCollectionOfListsTest extends TestsForSolutionMappingsIterableWithFilter
+{
+    @Test
+    public void oneList() {
+        final Collection<List<SolutionMapping>> solMapCollection = new ArrayList<>();
+        solMapCollection.add( getSolMapListWithTwoVar() );
+
+        // Iterate over all solution mappings contained in a collection
+        final Iterator<SolutionMapping> it = new SolutionMappingsIterableOverCollectionOfLists(solMapCollection).iterator();
+
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z1", var3 );
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z2", var3 );
+        assertHasNext( it, "http://example.org/y2", var2, "http://example.org/z3", var3 );
+        assertFalse( it.hasNext() );
+    }
+
+    @Test
+    public void twoListsSame() {
+        final Collection<List<SolutionMapping>> solMapCollection = new ArrayList<>();
+        solMapCollection.add( getSolMapListWithTwoVar() );
+        solMapCollection.add( getSolMapListWithTwoVar() );
+
+        // Iterate over all solution mappings contained in a collection
+        final Iterator<SolutionMapping> it = new SolutionMappingsIterableOverCollectionOfLists(solMapCollection).iterator();
+
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z1", var3 );
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z2", var3 );
+        assertHasNext( it, "http://example.org/y2", var2, "http://example.org/z3", var3 );
+
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z1", var3 );
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z2", var3 );
+        assertHasNext( it, "http://example.org/y2", var2, "http://example.org/z3", var3 );
+
+        assertFalse( it.hasNext() );
+    }
+
+    @Test
+    public void twoListsDifferent() {
+        final Collection<List<SolutionMapping>> solMapCollection = new ArrayList<>();
+        solMapCollection.add( getSolMapListWithTwoVar() );
+        solMapCollection.add( getSolMapListWithThreeVar() );
+
+        // Iterate over all solution mappings contained in a collection
+        final Iterator<SolutionMapping> it = new SolutionMappingsIterableOverCollectionOfLists(solMapCollection).iterator();
+
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z1", var3 );
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z2", var3 );
+        assertHasNext( it, "http://example.org/y2", var2, "http://example.org/z3", var3 );
+
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z1", var3, "http://example.org/x1", var1 );
+        assertHasNext( it, "http://example.org/y1", var2, "http://example.org/z2", var3, "http://example.org/x1", var1 );
+        assertHasNext( it, "http://example.org/y2", var2, "http://example.org/z2", var3, "http://example.org/x2", var1 );
```

Comment:
@chengsijin0817 Please take a look at what I did here. That makes reading the tests much easier. Can you please do the same in the other test classes here in this PR.

Reply:
@hartig Thanks a lot for your work. 
I have updated the other test classes according to your suggestions.